### ISSUE TITLE
Unexpected behavior using 'when'?

### ISSUE THREAD

**Issue by albizures** [enquirer/prompt-list#3]
Hi, I'm working with a bunch of list questions using the `when` function, but I'm facing a weird behavior.
For example:
```javascript
const questions = [{
  name: 'order',
  type: 'list',
  message: 'What would you like to order?',
  // choices may be defined as an array or a function that returns an array
  choices: [
    'Coke',
    'Diet Coke',
    'Cherry Coke',
    {name: 'Sprite', disabled: 'Temporarily unavailable'},
    'Water'
  ],
  when: function () {
    return false
  }
}]

enquirer
   .ask(questions)
   .then(answers => {
     console.log(answers)
   })
   .catch(err => {
     console.log(err)
   })
```
and the answers object will be:
```js
{ order: 'Coke' }
```
this is unexpected at least for me, I was expecting the order's answer to be `undefined`
so, I'm not sure if this is the right behavior and if it is, how can I avoid this case?

Thanks

**Comment by doowb** [comment]
Hi @albizures, thanks for the issue.

**TL;DR;**

As a workaround, you can set `this.position` to `-1` inside the `.when` function:

```js
  when: function() {
    this.position = -1;
    return false;
  }
```

Internally, when the `.when` function returns `false`, then answer is retrieved from the prompt and passed along without running the prompt. This allows default answers to be populated and returned so they can still be used like normal.

Since `prompt-list`'s `getAnswer` method uses `this.position` to lookup the selected item in the list, it's getting the first item `0` since no default was set. You can override this by changing `this.position` to an index that's not in the choices list (e.g. `-1`).

We could do something in `prompt-list` to override `.when` to ensure the original default value is returned when `.default` is set and `.when` returns `false`, otherwise set `this.position` to `-1` so `undefined` is returned. I think this makes more sense and will be expected.

Does this make sense? I'll try to put together a PR with what I'm thinking unless you want to do it :)

**Comment by albizures** [comment]
Sure, I'll do it.

Thanks.
